josh-proxy/src/bin/josh-proxy.rs

```diff
@@ -678,10 +678,15 @@ async fn run_proxy() -> josh::JoshResult<i32> {
                     error_response().await
                 };
                 let _e = s.enter();
+                let http_status = r.status().as_u16()
+                let log_level = match http_status {
+                    200 => tracing::Level::TRACE,
```

Comment:
I would prefer something like:
```
< 400 -> TRACE // All different kinds of success
< 500 -> WARN // Something wrong with the request, clients fault
_ -> ERROR // Server side error, need to debug josh
```

Reply:
Actually, the first should probably be <= 401, as "unauthorized" is a very common and normal response that will happen pretty much all the time without any need for concern.

Reply:
@christian-schilling but it also returns 500 here when checkout URL is malformed

Reply:
ah I think it still makes sense. Thanks, I'll update the PR a bit later

Reply:
If it returns 500 when the URL is malformed that's a bug in itself, it should be something 4xx.

Reply:
![image](https://user-images.githubusercontent.com/2057191/147762187-4c850204-06f5-459a-b76c-31e2b104244b.png)